erasurecoding/src/lib.rs

```diff
@@ -0,0 +1,122 @@
+use libc::{c_int, size_t};
+use std::slice;
+
+/// Reed-Solomon encode function for FFI
+/// Takes sharded original data as byte array and produces recovery shards
+/// Parameters:
+/// - original_shards: number of original data shards
+/// - recovery_shards: number of recovery shards to generate
+/// - original_shards: input byte array containing flattened shards
+/// - shard_size: size of each shard in bytes
+/// - recovery_shards_out: buffer to store generated and flattened recovery shards
+/// Returns 0 on success, -1 on error
+#[no_mangle]
+pub unsafe extern "C" fn reed_solomon_encode(
+    original_shards_count: size_t,
+    recovery_shards_count: size_t,
+    shard_size: size_t,
+    original_shards: *const u8,
+    recovery_shards_out: *mut u8,
+) -> c_int {
+    let original_shards: Vec<_> = slice::from_raw_parts(
+        original_shards,
+        original_shards_count as usize * shard_size as usize,
```

Comment:
Are we sure there won't be overflow?
If there might be, let's use, e.g. `checked_mul()` to handle that.

Reply:
I think we can do some extra checks on `original_shards_count` , which shouldn't be more than 65535. Also going to add `original_shards_len; so we can verify that the `shard_size` is correct. That should avoid an overflow.